modules/scripts/startup-script/main.tf

```diff
@@ -84,20 +75,20 @@ resource "random_id" "resource_name_suffix" {
 }
 
 module "config_storage_bucket" {
-  count                       = var.gcs_bucket_full_path == null ? 1 : 0
+  count                       = var.gcs_bucket_path == null ? 1 : 0
   source                      = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
   version                     = "~> 3.4"
 
   project_id                  = var.project_id
-  name                        = local.storage_bucket_name
+  name                        = "${var.deployment_name}-startup-scripts-${random_id.resource_name_suffix.hex}"
   location                    = var.region
   storage_class               = "REGIONAL"
   labels                      = var.labels
 }
 
 data "google_storage_bucket" "existing_bucket" {
-  count                       = var.gcs_bucket_full_path != null ? 1 : 0
-  name                        = local.storage_bucket_name
+  count                       = var.gcs_bucket_path != null ? 1 : 0
+  name                        = regex("^gs://([^/]*)/*(.*)", local.gcs_bucket_path_trimmed)[0]
```

Comment:
nit: You could store the regex string as a local constant since you use it twice. This way if it ever needs to be updated we won't forget to update it in both places (DRY).
```
locals {
  bucket_regex = "^gs://([^/]*)/*(.*)"
}
...
name = regex(local.bucket_regex, local.gcs_bucket_path_trimmed)[0]
```

Reply:
done.